src/engine/renderer/tr_shader.cpp

```diff
@@ -6421,49 +6435,46 @@ void R_ListShaders_f()
 		stageCount += shader->numStages;
 		highestShaderStageCount = std::max( highestShaderStageCount, shader->numStages );
 
-		std::string foundShaderTypeString = shaderTypeStrings[ shader->type ];
-		std::string foundShaderSortString = shaderSortStrings[ (shaderSort_t) shader->sort ];
-
-		std::string foundInteractLightString = shader->interactLight ? "INTERACTLIGHT" : "";
-
-		std::string shaderNameString = shader->name;
-		shaderNameString += shader->defaultShader ? " (DEFAULTED)" : "";
+		num = Str::Format( "%i", i );
+		shaderType = shaderTypeName[ shader->type ];
+		shaderSort = shaderSortName[ (shaderSort_t) shader->sort ];
+		interactLight = shader->interactLight ? "INTERACTLIGHT" : "";
+		shaderName = shader->name;
+		shaderName += shader->defaultShader ? " (DEFAULTED)" : "";
 
 		for ( int j = 0; j < shader->numStages; j++ )
 		{
 			shaderStage_t *stage = shader->stages[ j ];
 
-			std::string foundStageTypeString = stageTypeStrings[ stage->type ];
+			stageNumber = Str::Format( "%i", j );
```

Comment:
`stageNumber` and `num` don't need to be strings. You can send integers to `operator<<`

Reply:
Done.